File: com.python-unity-shared-memory/Tests/Editor/Test_Integration.cs
using NUnit.Framework;
using UnityEngine;
using System.IO;
using PythonUnitySharedMemory;

namespace PythonUnitySharedMemory.Tests.Editor{
   
    public class TestSharedMemoryIntegration
    {

        [Test]
        public void TestIntegration()
        {
            SharedMemory.DeleteFiles("test");
            Debug.Log("Launch the pytest command now, the test has started.");
            // <Block u0>
            var sm = new SharedMemory("test", 100, true, 30);
            sm.WriteString(42, "foo");
            // </block u0>
            sm.GiveControl(); //block p1
            //<block u2>
            Assert.AreEqual("bar", sm.ReadString(42).Item1);
            sm.WriteString(42, "foo");
            //</block u2>
            sm.GiveControl(); // block p3

            // <block u4>
            Assert.AreEqual("bar142", sm.ReadString(142).Item1);
            Assert.AreEqual("foo", sm.ReadString(42).Item1);


            sm.Resize(300);
            // </block u4>
            sm.GiveControl(); // block p5>

            //<block u6>
            Assert.AreEqual("bar342", sm.ReadString(342).Item1);
            sm.Dispose();
            // </block u6>
        }
    }
}

File: python_unity_shared_memory/test/test_integration.py
from python_unity_shared_memory import  SharedMemory, delete_files
import os
import pytest

def test_integration():
    sm = SharedMemory("test", 100, False, 30)
    # <block p1>
    assert sm.read_string(42)[0] == "foo"
    sm.write_string(42, "bar")
    assert sm.read_string(42)[0] == "bar"
    # </block p1>
    sm.give_control() # bock u2
    # <block p3>
    assert sm.read_string(42)[0] == "foo"

    sm.resize(200)
    sm.write_string(142, "bar142")
    #</block p3>
    sm.give_control() # block u4
    #<block p5>
    sm.resize(400)
    sm.write_string(342, "bar342")
    #</block p5>
    sm.give_control(wait=False) # block u6

